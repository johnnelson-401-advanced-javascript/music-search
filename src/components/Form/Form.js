import React from 'react';
import PropTypes from 'prop-types';

export default function SearchForm({ handleSubmit, handleChange, searchQuery }) {
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" name="searchQuery" value={searchQuery} onChange={handleChange} placeholder="Search Artists"></input>
      <button>Submit</button>
    </form>
  );
}
SearchForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  searchQuery: PropTypes.string.isRequired
};
