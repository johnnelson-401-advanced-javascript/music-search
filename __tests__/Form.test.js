import React from 'react';
import { shallow } from 'enzyme';
import Form from '../src/components/Form/Form';

describe('Form component', () => {
  it('renders Form', () => {
    const handleSubmit = jest.fn();
    const handleChange = jest.fn();
    const searchQuery = 'string';
    const wrapper = shallow(<Form 
      handleChange={handleChange} 
      handleSubmit={handleSubmit}
      searchQuery={searchQuery} />);
    expect(wrapper).toMatchSnapshot();
  });
});
