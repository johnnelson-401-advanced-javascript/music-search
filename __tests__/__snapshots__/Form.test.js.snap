// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <input
    name="searchQuery"
    onChange={[MockFunction]}
    placeholder="Search Artists"
    type="text"
    value="string"
  />
  <button>
    Submit
  </button>
</form>
`;
